src/pharmpy/modeling/parameter_sampling.py

```diff
@@ -181,8 +184,12 @@ def sample_parameters_uniformly(
     def fn(pe, lower, upper, n, rng):
         samples = np.empty((n, len(lower)))
         for i, (x, a, b) in enumerate(zip(pe, lower, upper)):
-            lower = max(a, x - x * fraction)
-            upper = min(b, x + x * fraction)
+            if scale == 'normal':
+                lower = max(a, x - x * fraction)
+                upper = min(b, x + x * fraction)
+            elif scale == 'UCP':
+                lower = 0.1 - 0.1 * fraction
+                upper = 0.1 + 0.1 * fraction
```

Comment:
For sampling on UCP scale don't we still want the resulting parameter samples on the normal scale?

Reply:
Or perhaps it is faster to do conversion later.